### DISCUSSION TITLE
Change in v2.5.2 causes checkboxes to break

### DISCUSSION THREAD

**Discussion by ahinkle** [livewire/livewire#3536] [7. Bugs (archived)]
Hey there, Livewire community! 👋 

I've been battling this one with a few of my applications. My tests suite caught that it's trying to save a checkbox input value as null when the initial state is false.

The issue started in v2.5.2 and has been a problem ever since - not allowing me to upgrade. I'm able to replicate the issue in my application with seeing the null result vs. bool

![Screen Shot 2021-08-17 at 7 57 30 AM](https://user-images.githubusercontent.com/17038330/129730078-3d8bb71d-70d7-4b31-9ba8-eb2577fcd3eb.png)

Replicate:

1. Create a component
2. Add a property and the save method to the component
```php
/**
 * Determine if the Billing address is the same as the Shipping address.
 *
 * @var bool
 */
public $sameAsBilling = false;

public function save()
{
    dd($this->sameAsBilling); // returns null when it should return true.
}
```
3. Add the field to your blade component:
```blade
<form wire:submit.prevent="submit">
    <input type="checkbox" wire:model="sameAsBilling" />
    <button type="submit">Submit</button>
</form>
```

With some validation rules and assertions, it's expecting a boolean but instead, it's receiving as `null`.  Anyone else having this issue? I'm still on 2.5.1 but when I upgrade to 2.5.2 is when this issue arises. 

Thanks!

**Comment by ahinkle** [comment]
Duplicate of #3305. 👍

**Reply by joshhanley** [reply]
@ahinkle can you expand on why you think it's a duplicate? That discussion was to do with virtual properties, but yours is checkboxes, so just want to confirm.

But that discussion was patched in #3320 and should be released with the next version.

If you get the chance, can you install `dev-master` and test to see if the issue has been fixed for you?

Thanks!
